scipy/spatial/transform/_rotation.pyx

```diff
@@ -1718,6 +1984,172 @@ cdef class Rotation:
+               PLoS ONE 17(11): e0276302. 10.1371/journal.pone.0276302
+        .. [3] https://en.wikipedia.org/wiki/Gimbal_lock#In_applied_mathematics
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Davenport angles are a generalization of Euler angles, when we use the
+        canonical basis axes:
+
+        >>> ex = [1, 0, 0]
+        >>> ey = [0, 1, 0]
+        >>> ez = [0, 0, 1]
+
+        Represent a single rotation:
+
+        >>> r = R.from_rotvec([0, 0, np.pi/2])
+        >>> r.as_davenport([ez, ex, ey], degrees=True)
+        array([90.,  0.,  0.])
+        >>> r.as_euler('zxy', degrees=True)
+        array([90.,  0.,  0.])
+        >>> r.as_davenport([ez, ex, ey], degrees=True).shape
+        (3,)
+
+        Represent a stack of single rotation:
+
+        >>> r = R.from_rotvec([[0, 0, np.pi/2]])
+        >>> r.as_davenport([ez, ex, ey], degrees=True)
+        array([[90.,  0.,  0.]])
+        >>> r.as_davenport([ez, ex, ey], degrees=True).shape
+        (1, 3)
+
+        Represent multiple rotations in a single object:
+
+        >>> r = R.from_rotvec([
+        ... [0, 0, 90],
+        ... [45, 0, 0]], degrees=True)
+        >>> r.as_davenport([ez, ex, ey], degrees=True)
+        array([[90.,  0.,  0.],
+               [ 0., 45.,  0.]])
+        >>> r.as_davenport([ez, ex, ey], degrees=True).shape
+        (2, 3)
+
+        Compared to Euler angles with asymmetrical sequences, sometimes the
+        returned angles are in a different set:
+
+        >>> r = R.from_euler('zyx', [0, 0, 90], degrees=True)
+        >>> euler = r.as_euler('zyx', degrees=True)
+        >>> daven = r.as_davenport([ez, ey, ex], degrees=True)
+        >>> euler
+        array([ 0.,  0., 90.])
+        >>> daven
+        array([180., 180., -90.])
+
+        But they still represent the same rotation (in this example, with some
+        low numerical errors):
+
+        >>> r = R.from_euler('zyx', [180., 180., -90.], degrees=True)
+        >>> r.as_euler('zyx', degrees=True)
+        array([-1.27222187e-14,  1.27222187e-14,  9.00000000e+01])
+
+        In order to get the same set of angles, the option ``correct_set`` must
+        be set to ``True``:
+
+        >>> r = R.from_euler('zyx', [0, 0, 90], degrees=True)
+        >>> euler = r.as_euler('zyx', degrees=True)
+        >>> daven = r.as_davenport([ez, ey, ex], degrees=True, correct_set=True)
+        >>> euler
+        array([ 0.,  0., 90.])
+        >>> daven
+        array([-1.27222187e-14, -0.00000000e+00,  9.00000000e+01])
+
+        """
+        if len(axes) != 3:
+            raise ValueError("Expected 3 axes, got {}.".format(len(axes)))
+
+        for axis in axes:
+            if len(axes) != 3:
+                raise ValueError("Axes must be vectors of length 3.")
+
+        n1, n2, n3 = axes
+
+        if np.dot(n1, n2) >= 1e-7:
```

Comment:
I think these checks should be made after normalization (to avoid dependency on the norm) and using `or` to combine both checks looks like a good idea.